pulp_ansible/tests/functional/api/test_import.py

```diff
@@ -0,0 +1,169 @@
+        """Set database to a clean state."""
+        self.import_repos = []
+        self._clean_database()
+        self._setup_repositories(self.import_repos)
+        self.importer_api = ImportersPulpApi(self.core_client)
+        self.imports_api = ImportersCoreImportsApi(self.core_client)
+        self.tags_api = PulpAnsibleTagsApi(self.client)
+        self.croles_api = ContentRolesApi(self.client)
+        self.repo_ver_api = RepositoriesAnsibleVersionsApi(self.client)
+
+    def _create_importer(self, name=None, cleanup=True):
+        """Create an importer."""
+        mapping = {}
+        if not name:
+            name = uuid4()
+
+        for idx, repo in enumerate(self.export_repos):
+            mapping[repo.name] = self.import_repos[idx].name
+
+        body = {
+            "name": name,
+            "repo_mapping": mapping,
+        }
+
+        importer = self.importer_api.create(body)
+
+        if cleanup:
+            self.addCleanup(self.importer_api.delete, importer.pulp_href)
+
+        return importer
+
+    def _perform_import(self, importer, chunked=False):
+        """Perform an import with importer."""
+        if chunked:
+            filenames = [
+                f for f in list(self.chunked_export.output_file_info.keys()) if f.endswith("json")
+            ]
+            import_response = self.imports_api.create(importer.pulp_href, {"toc": filenames[0]})
+        else:
+            filenames = [
+                f for f in list(self.export.output_file_info.keys()) if f.endswith("tar.gz")
+            ]
+            import_response = self.imports_api.create(importer.pulp_href, {"path": filenames[0]})
+        monitor_task(import_response.task)
+        task = self.api_client.get(import_response.task)
+        resources = task["created_resources"]
+        task_group_href = resources[1]
+        task_group = monitor_task_group(task_group_href)
+
+        return task_group
+
+
+class PulpImportTestCase(BaseImport):
+    """
+    Basic tests for PulpImporter and PulpImport.
+    """
+
+    def test_import(self):
+        """Test an import."""
+        self._setup_import()
+        importer = self._create_importer()
+        task_group = self._perform_import(importer)
+        self.assertEqual(len(self.import_repos) + 1, task_group.completed)
+        for repo in self.import_repos:
+            repo = self.repo_api.read(repo.pulp_href)
+            self.assertEqual(f"{repo.pulp_href}versions/1/", repo.latest_version_href)
+            repo_ver = self.repo_ver_api.read(repo.latest_version_href)
+            added = repo_ver.content_summary.added
+            if "ansible.role" in added.keys():
+                roles = self.croles_api.list()
+                self.assertEqual(roles.count, added["ansible.role"]["count"])
+            elif "ansible.collection_version" in added.keys():
+                cvs = self.cv_api.list()
+                self.assertEqual(cvs.count, added["ansible.collection_version"]["count"])
```

Comment:
It looks like this code is checking that the content that got added to the repo version is equal to the number of content is the system? It seems like it would be more useful to compare the content count from the exported repo(s) to repo_ver.content_summary.added.